packages/backend-function/src/schedule_parser.ts

```diff
@@ -0,0 +1,209 @@
+import { CronOptions, Rule, Schedule } from 'aws-cdk-lib/aws-events';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as targets from 'aws-cdk-lib/aws-events-targets';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+import type { Cron, Rate, TimeInterval } from './factory.js';
+
+/**
+ * Parses function schedule props in order to create EventBridge rules.
+ */
+export class ScheduleParser {
+  /**
+   * Initialize EventBridge rules
+   */
+  constructor(
+    private readonly lambda: NodejsFunction,
+    private readonly timeIntervals: TimeInterval[]
+  ) {
+    this.timeIntervals.forEach((interval, index) => {
+      if (isRate(interval)) {
+        const { value, unit } = parseRate(interval);
+
+        if (value && !isPositiveWholeNumber(value)) {
+          throw new Error(`schedule must be set with a positive whole number`);
+        }
+
+        if (
+          value &&
+          lambda.timeout &&
+          unit === 'm' &&
+          value * 60 < lambda.timeout.toSeconds()
+        ) {
+          const timeout = lambda.timeout.toSeconds();
+
+          throw new Error(
+            `schedule must be greater than the timeout of ${timeout} ${
+              timeout > 1 ? 'seconds' : 'second'
+            }`
+          );
+        }
+      } else {
+        if (!isValidCron(interval)) {
+          // TODO: Better error messaging here or throw within each part of isValidCron in order to give more concise error messages
+          throw new Error(`schedule cron expression is not valid`);
+        }
+      }
+
+      try {
+        // Lambda name will be prepended to rule id, so only using index here for uniqueness
+        const rule = new Rule(this.lambda, `lambda-schedule${index}`, {
+          schedule: Schedule.cron(translateToCronOptions(interval)),
+        });
+
+        rule.addTarget(new targets.LambdaFunction(this.lambda));
+      } catch (error) {
+        throw new AmplifyUserError(
+          'NodeJSFunctionScheduleInitializationError',
+          {
+            message: 'Failed to instantiate schedule for nodejs function',
+            resolution: 'See the underlying error message for more details.',
+          },
+          error as Error
+        );
+      }
+    });
+  }
+}
+
+const isRate = (timeInterval: TimeInterval): timeInterval is Rate => {
+  return timeInterval.split(' ')[0] === 'every';
```

Comment:
More exhaustive validation
```suggestion
  const parts = timeInterval.split(' ');
  return parts[0] === 'every' && (['m', 'h', 'day', 'week', 'year', 'month'].some(a => parts[1].endsWith(a))) && parts.length === 2
```

Because customers could be using this in a JS only app or have TS errors suppressed in their files or just creative enough to pass `every 5 weeks until 2025` (Which our type would allow thinking it's a cron)

Reply:
We can also get rid of it by just using `tryParseRate`.